quickwit-search/src/client_pool/search_client_pool.rs

```diff
@@ -177,6 +208,185 @@ impl ClientPool for SearchClientPool {
                 error!("Missing client. This should never happen! Please report");
             }
         }
+
         Ok(client_to_jobs)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::net::{SocketAddr, TcpListener};
+    use std::sync::Arc;
+    use std::thread;
+    use std::time;
+
+    use quickwit_cluster::cluster::{read_host_key, Cluster, EpidemicCluster};
+    use quickwit_common::to_socket_addr;
+
+    use crate::client_pool::search_client_pool::create_search_service_client;
+    use crate::client_pool::{ClientPool, Job};
+    use crate::swim_addr_to_grpc_addr;
+    use crate::SearchClientPool;
+
+    #[tokio::test]
+    async fn test_search_client_pool_single_node() -> anyhow::Result<()> {
+        let tmp_dir = tempfile::tempdir()?;
+
+        let host_key_path = tmp_dir.path().join("host_key");
+        let host_key = read_host_key(host_key_path.as_path())?;
+        let tmp_swim_port = available_port()?;
+        let swim_addr_string = format!("127.0.0.1:{}", tmp_swim_port);
+        let swim_addr = to_socket_addr(&swim_addr_string)?;
+        let cluster = Arc::new(EpidemicCluster::new(host_key, swim_addr)?);
```

Comment:
helper function

Reply:
I changed the code a bit. Unnecessary variables have been removed.
It's not a helper function, but these variables are used in the assertions that follow. How about this one?